storage/src/archive/immutable/mod.rs

```diff
@@ -47,8 +48,8 @@
 //!         freezer_journal_compression: Some(3),
 //!         ordinal_partition: "ordinal".into(),
 //!         items_per_section: 1024,
```

Comment:
This change makes me wonder if we should require all numeric items in these configs to be NonZero?

The reason I originally held off on this change is because asserting the same level of checks on all args would be a meaningful lift from where we are (basically non-zero numbers and non-zero length strings).

Wondering if it would just be easier to assert for now to avoid having all args/configs everywhere need to use this macro 🤔.

I "think" this is most proper so we "should" do it this way but may be a bit of a journey to get it fully applied.

Reply:
That being said, I suppose the most coherent way to go across the larger migration is to just go crate by crate like this (assuming we all agree the NonZero* is the end goal).

Reply:
For now yeah I'm just doing the buffer capacity vars, but does seem we could change a bunch of others as well.